common/http/platform/beast/http.cpp

```diff
@@ -565,26 +615,37 @@ void Client::ReadHeaderHandler(const error_code &ec, size_t num_read) {
 		return;
 	}
 
-	if (response_ && !response_->body_async_reader_) {
-		// If there is no registered reader, then we need to schedule the download
-		// ourselves. Else the reader will do it.
-		ReadNextBodyPart(body_buffer_.size());
+	if (body_status_ == BodyReadingStatus::None) {
+		CallErrorHandler(MakeError(BodyIgnoredError, ""), request_, body_handler_);
 	}
 }
 
 void Client::AsyncReadNextBodyPart(
 	vector<uint8_t>::iterator start, vector<uint8_t>::iterator end, io::AsyncIoHandler handler) {
+	if (body_status_ == BodyReadingStatus::None
+		|| body_status_ == BodyReadingStatus::ReaderCreated) {
+		body_status_ = BodyReadingStatus::InProgress;
+	}
+
+	if (body_status_ != BodyReadingStatus::InProgress) {
+		handler(0);
+		if (body_status_ == BodyReadingStatus::ReachedEnd) {
```

Comment:
I don't get it. Otherwise is an error and we should also call the user body handler, right?

Reply:
This is called by the reader, which should always return `0` even if called multiple times. The body handler however, should only be called once, which is why we have this block to determine whether we have already done so or not.